ts/sagas/startup.ts

```diff
@@ -78,15 +80,17 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
   // FIXME: since it looks like we load the session info every
   //        time we get a session token, think about merging the
   //        two steps.
-  const maybeSessionInformation: ReturnType<
+  // tslint:disable-next-line: no-let
+  let maybeSessionInformation: ReturnType<
     typeof sessionInfoSelector
   > = yield select(sessionInfoSelector);
   if (isSessionRefreshed || maybeSessionInformation.isNone()) {
     // let's try to load the session information from the backend.
-    const maybeSessionInfo: SagaCallReturnType<
-      typeof loadSessionInformationSaga
-    > = yield call(loadSessionInformationSaga, backendClient.getSession);
-    if (maybeSessionInfo.isNone()) {
+    maybeSessionInformation = yield call(
+      loadSessionInformationSaga,
+      backendClient.getSession
+    );
+    if (maybeSessionInformation.isNone()) {
```

Comment:
if you change this check (and the one at row 84) to `isNone(maybeSessionInformation)`, the compiler should be able to infer that, after the `if` block, `maybeSessionInformation` is a `some`, avoiding the `.isSome()` check down the line - the exercise here is to make sure we have a valid session info after this block (and having the compiler be aware of it)

Reply:
`maybeSessionInformation.isNone()` should correctly type guard `maybeSessionInformation` (see [here](https://github.com/gcanti/fp-ts/blob/fd0ccf39bb52622f93e01fb7b9bcb6c955d634fe/src/Option.ts#L377)), so the compiler should already infer the type correctly. Indeed, the `isSome` check later on is superfluous and can be safely removed (I added it *before* changing this part here and forgot to get rid of it afterwards)